src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { createExternalReference } from 'tests/fixtures/entity-viewer/product';
+import { ExternalSource } from 'src/content/app/entity-viewer/shared/helpers/entity-helpers';
+
+jest.mock(
+  'src/shared/components/instant-download/instant-download-protein/InstantDownloadProtein',
+  () => jest.fn()
+);
+
+const swissprotXref = createExternalReference({
+  accession_id: 'SWISS123',
+  source: {
+    name: 'UniProtKB/Swiss-Prot',
+    id: 'swiss_123'
+  }
+});
+
+const tremblSource = {
+  name: 'UniProtKB/TrEMBL',
+  id: 'trembl1'
+};
+const tremblXrefs = times(4, () =>
+  createExternalReference({ source: tremblSource })
+);
+
+times(4, (index) => (tremblXrefs[index].accession_id = `trembl_${index}`));
+
+describe('<ProteinsListItemInfo /', () => {
+  it('renders a single xref', () => {
+    const props = {
+      source: ExternalSource.UNIPROT_SWISSPROT,
+      xrefs: [swissprotXref]
+    };
+    // ProteinExternalReferenceGroup expect xrefs from a single source
+    const { queryByText } = render(
+      <ProteinExternalReferenceGroup {...props} />
+    );
+    expect(queryByText(swissprotXref.source.name)).toBeTruthy();
+    expect(queryByText(swissprotXref.accession_id)).toBeTruthy();
+  });
+
+  describe('multiple xrefs', () => {
+    it('renders 2 xrefs', () => {
+      const props = {
+        source: ExternalSource.UNIPROT_TREMBL,
+        xrefs: tremblXrefs.slice(0, 2)
+      };
+      const { queryByText, queryAllByText } = render(
+        <ProteinExternalReferenceGroup {...props} />
+      );
+      expect(queryAllByText(props.source)).toHaveLength(2);
+      expect(queryByText(props.xrefs[0].accession_id)).toBeTruthy();
+      expect(queryByText(props.xrefs[1].accession_id)).toBeTruthy();
+    });
+
+    it('displays 1 xref with an option to expand other ids if there are more than 3 xrefs', () => {
+      const props = {
+        source: ExternalSource.UNIPROT_TREMBL,
+        xrefs: tremblXrefs
+      };
+      const { container, queryByText, queryAllByText } = render(
+        <ProteinExternalReferenceGroup {...props} />
+      );
+      expect(queryByText(props.source)).toBeTruthy();
+      expect(queryAllByText(props.xrefs[0].accession_id)).toHaveLength(1);
+      const chevron = container.querySelector('.chevron');
+      expect(chevron).toBeTruthy();
+    });
+
+    it('displays all xrefs when expanded if there are more than 3 xrefs', () => {
+      const props = {
+        source: ExternalSource.UNIPROT_TREMBL,
+        xrefs: tremblXrefs
+      };
+      const { container, getByText, getAllByText } = render(
+        <ProteinExternalReferenceGroup {...props} />
+      );
+      const chevron = container.querySelector('.chevron') as HTMLElement;
+      userEvent.click(chevron);
+      expect(getByText(props.xrefs[1].accession_id)).toBeTruthy();
+      expect(getAllByText(props.source)).toHaveLength(4);
+      expect(container.querySelectorAll('.externalLinkContainer')).toHaveLength(
+        4
+      );
+    });
```

Comment:
It might be worth combining the above two tests into one (they are effectively two parts of the same test case, and their setup part is identical), in which the first half of the test will check the expectations before you click on the chevron, and the second half will check the expectations after you click on the chevron.